docs/docs/guides/build/assets/asset-selection-syntax.md

```diff
@@ -254,98 +250,281 @@ manhattan_job = define_asset_job(name="manhattan_job", selection="++manhattan_ma
+taxi_jobs_two_downstream = AssetSelection.from_string("taxi_zones_file+2")
 ```
 
 </TabItem>
 <TabItem value="cli" label="CLI">
 
-When selecting an asset's dependencies using the CLI, enclose the asterisk (`*`) and the asset key in double quotes (`"`):
+When selecting an asset's dependencies using the CLI, enclose the number and plus sign (`+`) and the asset key in double quotes (`"`):
 
 ```shell
-dagster asset list --select "taxi_zones_file*"
-dagster asset materialize --select "taxi_zones_file*"
+dagster asset list --select "taxi_zones_file+2"
+dagster asset materialize --select "taxi_zones_file+2"
 ```
 
 </TabItem>
 <TabItem value="dagster-ui" label="Dagster UI">
 
 ```shell
-taxi_zones_file*
+taxi_zones_file++
 ```
 
 Which would result in the following asset graph:
 
-![Screenshot of Daggy U project graph](/images/guides/build/assets/asset-selection-syntax/select-downstream-dependencies.png)
+![TODO]()
 
 </TabItem>
 </Tabs>
 
-#### Selecting a specific number of downstream layers \{#specific-downstream}
+### Select all upstream and downstream layers \{#full-lineage}
 
-To select an asset and multiple downstream layers, add plus sign (`+`) for each layer you want to select after the asset key in the query.
+To select an asset's entire lineage (all upstream and downstream layeres), add a plus sign (`+`) before and after the asset key in the query.
 
-This example selects the `taxi_trips_file` asset and two downstream layers.
+This example selects the entire lineage for the `taxi_zones` asset.
 
-<Tabs>
+<Tabs groupId="examples">
 <TabItem value="python" label="Python">
 
 ```python
-taxi_zones_job = define_asset_job(name="taxi_zones_job", selection="taxi_zones_file++")
+taxi_zones_all = AssetSelection.from_string("+taxi_zones+")
 ```
 
 </TabItem>
 <TabItem value="cli" label="CLI">
 
-When selecting an asset's dependencies using the CLI, enclose the plus sign (`+`) and the asset key in double quotes (`"`):
+When selecting an asset's entire lineage using the CLI, enclose the plus sign (`+`) and the asset key in double quotes (`"`):
 
 ```shell
-dagster asset list --select "taxi_zones_file++"
-dagster asset materialize --select "taxi_zones_file++"
+dagster asset list --select "+taxi_zones+"
+dagster asset materialize --select "+taxi_zones+"
 ```
 
 </TabItem>
 <TabItem value="dagster-ui" label="Dagster UI">
 
 ```shell
-taxi_zones_file++
++taxi_zones+
 ```
 
 Which would result in the following asset graph:
 
-![Screenshot of Daggy U project graph](/images/guides/build/assets/asset-selection-syntax/select-two-downstream-layers.png)
+![TODO]()
 
 </TabItem>
 </Tabs>
+
+## Filters, operands, and grouping
+
+Filters allow you to narrow your asset selection using specific criteria.
+
+| Filter | Syntax | Description |
+|--------|--------|-------------|
+| **Exact key** | `key:"my_key"` | Selects assets with the exact key `my_key`. |
+| **Key substring** | `key_substring:"partial_key"` | Selects assets whose key contains `partial_key`. |
```

Comment:
We remove `key_substring`, instead we're allowing wildcards for the `key` filter only.

For example:

`key:prefix_*` would select all assets with keys starting with `prefix_`. 
You can also have multiple wildcards:

`key:prefix_*_middlefix_*_suffix*`